### ISSUE TITLE
Liquid effects remaining on Humans after the responsible liquid is fully filtered out of there blood and put back into them.

### ISSUE THREAD

**Issue by Nattie98** [studio-minus/ppg-bugs-and-requests#1373]
**Describe the bug**
Whenever you put Oil or Nitroglycerine into a Human the effects of that liquid remain even if you filter all of the responsible liquid out of them and put back in the blood you filtered said bad liquids out they still suffer from the effects of the bad liquid. Filter out all the Nitroglycerine they still Explode on high impact even though there is none in there filtered blood, filter out all the Oil, and there heart still stops even if you use a Cure syringe after removing all the oil from there blood and putting it back into them.

**Clearly describe how to reproduce the bug**
1. Inject Oil/Nitroglycerine into a Human
2. Remove all there blood and run it through a Centrifuge
3. Insert the now clean blood back into them and revive them.
If its Oil there heart will still stop and if its Nitroglycerine they still explode when heavily hit or thrown to hard.

**Mod list**
While I do use the Liquid Related Machinery Mod to make filtering a Humans blood more convenient as found here. https://steamcommunity.com/sharedfiles/filedetails/?id=2550340103 I have tested this in vanilla with Liquid canisters, Flasks and a Centrifuge and it still occurs Mods or no 

**System info**
 - Operating system: Windows 10
 - Game version: 1.24.3 (64 Bit Version)
 - GPU: Radeon RX 5500 XT
 - CPU: AMD FX-6300 Six-Core Processor

Game Debug info if you need it - [exported game debug info.md](https://github.com/studio-minus/ppg-bugs-and-requests/files/8762692/exported.game.debug.info.md)

**Comment by Nattie98** [comment]
> Wouldn't you think having Oil in your blood-stream would cause permanent damages to your body?

Yes but the problem is the Oil is removed and a Cure Syringe would completely undo that so it makes no sense if you remove all the Oil and use that and still have him die. The permanent damage is undone and the cause of it is removed so why is he still dying?

**Comment by Nattie98** [comment]
I am actually closing the issue cause I found the problem the Liquid machinery Mods filtration is bad so even though it filters the liquid it doesn't get all of it if you don't get the liquid out of the filtration leaving a tiny tiny amount but because of the game seeing even a very microscopic amount it counts the effect on the person so I'm gonna close it but I will say raising liquid tolerances would be ideal since it would improve the room for error.

**Comment by Nattie98** [comment]
I thought about closing it if its really a liquid filtration issue but given its a liquid tolerance problem I am keeping it open till I get some feed back on this.

**Comment by mestiez** [comment]
I have fixed this bug, it was hidden very well by my absolutely perfect shitshow codebase. Thanks for reporting

**Comment by Nattie98** [comment]
> I have fixed this bug, it was hidden very well by my absolutely perfect shitshow codebase. Thanks for reporting

On one hand I was like is this is a filtration issue or is this a liquid tolerance issue I though about closing it initially since i was like its not representative of the filtration issue of the initial problem but then I though ill just leave it open and let you look into it to see what you think when you do. Im glad to help :)

I wouldn't call it a shit show with programming cause when you program a game they don't really write a book about your particular situation for all intents and purposes more often then not your trail blazing territory and amidst that as you build onto it things that seemed sound either sound more unsound or break from things later added. So its not that it was bad its that as the game evolved and your more inital lines of code don't hold up to time with the new additions.

Its the same thing that happened with Anton and the VR title Hotdogs, Horseshoes and Hand grenades he had to rewrite the Ai code cause its not that it was bad it just didn't hold up to his new additions to Ai pathfinding and its technicalities he added.
